include/ceed.h

```diff
@@ -242,11 +242,34 @@ typedef enum {
   CEED_GAUSS_LOBATTO = 1,
 } CeedQuadMode;
 
+/// Type of basis shape to create non-tensor H1 element basis
+/// @ingrounp CeedBasis
+typedef enum {
+  /// Line
+  CEED_LINE = 0,
+  /// Quadralateral - 2D shape
+  CEED_QUAD = 1,
+  /// Hexehedron - 3D shape
+  CEED_HEX = 2,
+  /// Triangle - 2D shape
+  CEED_TRIANGLE = 4,
+  /// Tetrahedron - 3D shape
+  CEED_TET = 8,
+  /// Pyramid - 3D shape
+  CEED_PYRAMID = 16,
+  /// Prism - 3D shape
+  CEED_PRISM = 32,
```

Comment:
What is the logic about numbering here? If you had, say, `dim << 16 | topo_type` then you could get the dimension with a uniform operation. Also, it looks like we don't actually care about the topology anywhere in the library, though in case of mixed meshes we'll need to identify "like topologies". I could imagine we eventually add a constructor so the user can create new topologies.

Reply:
I'll rethink the numbering. I like your suggestion of embedding the dim in the number.

Reply:
Updated. I am using the first three bits for dimension and the last three for shape. 

001xxx is 1D
010xxx is 2D
100xxx is 3D

Right now I don't see the library caring about the topology, but I imagine that some backends will.